File: java/integrations/java-gradle-lambda-cdk/lambda/src/main/java/my/example/LambdaHandler.java
/*
 * Copyright (c) 2024 - Restate Software, Inc., Restate GmbH
 *
 * This file is part of the Restate examples,
 * which is released under the MIT license.
 *
 * You can find a copy of the license in the file LICENSE
 * in the root directory of this repository or package or at
 * https://github.com/restatedev/examples/
 */

package my.example;

import dev.restate.sdk.endpoint.Endpoint;
import dev.restate.sdk.lambda.BaseRestateLambdaHandler;

import my.example.Greeter;

public class LambdaHandler extends BaseRestateLambdaHandler {
    @Override
    public void register(Endpoint.Builder builder) {
        builder.bind(new Greeter());
    }
}

File: kotlin/integrations/kotlin-gradle-lambda-cdk/lambda/src/main/kotlin/dev/restate/sdk/examples/LambdaHandler.kt
/*
 * Copyright (c) 2024 - Restate Software, Inc., Restate GmbH
 *
 * This file is part of the Restate examples,
 * which is released under the MIT license.
 *
 * You can find a copy of the license in the file LICENSE
 * in the root directory of this repository or package or at
 * https://github.com/restatedev/examples/
 */

 package dev.restate.sdk.examples

import dev.restate.sdk.endpoint.Endpoint
import dev.restate.sdk.lambda.BaseRestateLambdaHandler

class LambdaHandler : BaseRestateLambdaHandler() {
    override fun register(builder: Endpoint.Builder) {
        builder.bind(Greeter())
    }
}
